packages/bson-transpilers/symbols/php/templates.yaml

```diff
@@ -0,0 +1,1271 @@
+        }
+    LongSymbolMaxArgsTemplate: &LongSymbolMaxArgsTemplate null
+    LongSymbolMinTemplate: &LongSymbolMinTemplate !!js/function >
+        () => {
+            return 'PHP_INT_MIN';
+        }
+    LongSymbolMinArgsTemplate: &LongSymbolMinArgsTemplate null
+    LongSymbolZeroTemplate: &LongSymbolZeroTemplate !!js/function >
+        () => {
+            return '0';
+        }
+    LongSymbolZeroArgsTemplate: &LongSymbolZeroArgsTemplate null
+    LongSymbolOneTemplate: &LongSymbolOneTemplate !!js/function >
+        () => {
+            return '1';
+        }
+    LongSymbolOneArgsTemplate: &LongSymbolOneArgsTemplate null
+    LongSymbolNegOneTemplate: &LongSymbolNegOneTemplate !!js/function >
+        () => {
+            return '-1';
+        }
+    LongSymbolNegOneArgsTemplate: &LongSymbolNegOneArgsTemplate null
+    LongSymbolFromBitsTemplate: &LongSymbolFromBitsTemplate !!js/function >
+        () => {
+            return '';
+        }
+    LongSymbolFromBitsArgsTemplate: &LongSymbolFromBitsArgsTemplate null
+    LongSymbolFromIntTemplate: &LongSymbolFromIntTemplate !!js/function >
+        () => {
+            return '';
+        }
+    LongSymbolFromIntArgsTemplate: &LongSymbolFromIntArgsTemplate !!js/function >
+        (lhs, arg) => {
+            return arg;
+        }
+    LongSymbolFromNumberTemplate: &LongSymbolFromNumberTemplate !!js/function >
+        () => {
+            return '';
+        }
+    LongSymbolFromNumberArgsTemplate: &LongSymbolFromNumberArgsTemplate !!js/function >
+        (lhs, arg) => {
+            return `(int) ${arg}`;
+        }
+    LongSymbolFromStringTemplate: &LongSymbolFromStringTemplate !!js/function >
+        () => {
+            return '';
+        }
+    LongSymbolFromStringArgsTemplate: &LongSymbolFromStringArgsTemplate !!js/function >
+        (lhs, arg) => {
+            return `(int) ${arg}`;
+        }
+    Decimal128SymbolFromStringTemplate: &Decimal128SymbolFromStringTemplate !!js/function >
+        () => {
+            return 'new \\MongoDB\\BSON\\Decimal128';
+        }
+    Decimal128SymbolFromStringArgsTemplate: &Decimal128SymbolFromStringArgsTemplate !!js/function >
+        (lhs, arg) => {
+            return `(${arg})`;
+        }
+    ObjectIdCreateFromHexStringTemplate: &ObjectIdCreateFromHexStringTemplate !!js/function >
+        () => {
+            return 'new \\MongoDB\\BSON\\ObjectId';
+        }
+    ObjectIdCreateFromHexStringArgsTemplate: &ObjectIdCreateFromHexStringArgsTemplate !!js/function >
+        (lhs, arg) => {
+            return `(${arg})`;
+        }
+    ObjectIdCreateFromTimeTemplate: &ObjectIdCreateFromTimeTemplate !!js/function >
+        () => {
+            return 'new \\MongoDB\\BSON\\ObjectId';
+        }
+    ObjectIdCreateFromTimeArgsTemplate: &ObjectIdCreateFromTimeArgsTemplate !!js/function >
+        (lhs, arg, isNumber) => {
+            if (!isNumber) {
+                return `(str_pad(dechex((${arg})->getTimestamp()), 24, '0'))`;
```

Comment:
Is `isNumber` is false, what is the expected type of `arg`?

Note that the timestamp component of an [ObjectId is only four bytes](https://www.mongodb.com/docs/manual/reference/method/ObjectId/). It's quite possible that a timestamp from an actual date type might _exceed_ four bytes, in which case it would need to be truncated. As-is, this code is going to allow it to overflow into other ObjectId components.

Additionally, we must consider that the timestamp component of the ObjectId is big endian (despite BSON itself being little endian). The best way to generate this four-byte, big endian value is with [`pack`](https://www.php.net/manual/en/function.pack.php) (previous discussed in https://github.com/mongodb/mongo-php-driver/issues/868#issuecomment-436022565 towards the end of that comment). For example:

```
>>> new \MongoDB\BSON\ObjectId
=> MongoDB\BSON\ObjectId {#2666
     +"oid": "62aa4c0c937b33f2a70c0a31",
   }

>>> \bin2hex(\pack('N', time()))
=> "62aa4c0d"
```

From there, you can `str_pad` as needed.

Reply:
Fixed.

Reply:
OK to resolve.